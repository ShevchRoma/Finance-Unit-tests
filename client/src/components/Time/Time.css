.top__body {
  display: flex;
  justify-content: space-evenly;
}
.top__column {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
  border-radius: 10px;
}
.column-top__row {
  display: flex;
  align-items: center;
}
.column-top__row:first-child {
  margin-right: 20px;
}
.column-top__row:last-child {
  flex-direction: column;
}

.column-top__dot {
  width: 15px;
  height: 15px;
  background: rgb(44, 244, 9);
  border: 1px solid green;
  border-radius: 50%;
  margin-right: 5px;
}
