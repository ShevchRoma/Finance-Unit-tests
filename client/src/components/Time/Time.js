import React from "react";
import "./Time.css";

const TopTicker = () => {
  return (
    <div className="top__body">
      <div className="top__column column-top">
        <div className="column-top__row">
          <div className="column-top__dot"></div>
          <p className="column-top__name">
            15:40{/* <Moment format="HH:mm:ss" interval={1000} /> */}
          </p>
        </div>
        <div className="column-top__row">
          <div className="column-top__percent">Time</div>
          <div className="column-top__coefficient">Decor</div>
        </div>
      </div>
    </div>
  );
};

export default TopTicker;
