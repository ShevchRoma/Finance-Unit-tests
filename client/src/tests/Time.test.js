import { cleanup } from "@testing-library/react";
import Time from "../components/Time/Time";
import TestRenderer from "react-test-renderer";

afterEach(cleanup);

describe("Ticker.js", () => {
  it("snapshot renders correctly, truthy values", () => {
    const tree = TestRenderer.create(<Time />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("snapshot renders correctly, falsy values", () => {
    const tree = TestRenderer.create(<Time />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
