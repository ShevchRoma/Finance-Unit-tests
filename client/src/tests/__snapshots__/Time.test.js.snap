// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker.js snapshot renders correctly, falsy values 1`] = `
<div
  className="top__body"
>
  <div
    className="top__column column-top"
  >
    <div
      className="column-top__row"
    >
      <div
        className="column-top__dot"
      />
      <p
        className="column-top__name"
      >
        15:40
      </p>
    </div>
    <div
      className="column-top__row"
    >
      <div
        className="column-top__percent"
      >
        Time
      </div>
      <div
        className="column-top__coefficient"
      >
        Decor
      </div>
    </div>
  </div>
</div>
`;

exports[`Ticker.js snapshot renders correctly, truthy values 1`] = `
<div
  className="top__body"
>
  <div
    className="top__column column-top"
  >
    <div
      className="column-top__row"
    >
      <div
        className="column-top__dot"
      />
      <p
        className="column-top__name"
      >
        15:40
      </p>
    </div>
    <div
      className="column-top__row"
    >
      <div
        className="column-top__percent"
      >
        Time
      </div>
      <div
        className="column-top__coefficient"
      >
        Decor
      </div>
    </div>
  </div>
</div>
`;
